phoenix-core/src/main/java/org/apache/phoenix/calcite/rel/PhoenixLimit.java

```diff
@@ -76,20 +76,27 @@ public RelOptCost computeSelfCost(RelOptPlanner planner, RelMetadataQuery mq) {
     
     @Override 
     public double estimateRowCount(RelMetadataQuery mq) {
-        double rows = super.estimateRowCount(mq);        
+        double rows = super.estimateRowCount(mq);
+        if(offset != null) {
+            return Math.max(0, Math.min(RexLiteral.intValue(fetch), rows - RexLiteral.intValue(offset)));
+        }
         return Math.min(RexLiteral.intValue(fetch), rows);
     }
 
     @Override
     public QueryPlan implement(PhoenixRelImplementor implementor) {
         QueryPlan plan = implementor.visitInput(0, (PhoenixQueryRel) getInput());
         int fetchValue = RexLiteral.intValue(fetch);
```

Comment:
I believe now that we've enabled offset, "this.fetch" could be also be null, which means it should be handled the same way as this.offset now. Could you please also add test cases to cover situations like "offset != null" but "limit == null"?